app/services/gov_uk_notify_status_client.rb

```diff
@@ -0,0 +1,18 @@
+class GovUkNotifyStatusClient
+  def initialize
+    @client = Notifications::Client.new(ENV.fetch("NOTIFY_KEY", nil))
+  end
+
+  def get_email_notifications(options)
+    opts = options.merge(template_type: "email")
+    notifications = @client.get_notifications(opts).collection
+
+    Enumerator.new do |yielder|
+      while notifications.any?
+        notifications.each { |n| yielder << n }
+
+        notifications = @client.get_notifications(opts.merge(older_than: notifications.last.id)).collection
+      end
+    end
```

Comment:
I am struggling to understand what this code does.

Is this the way to `Fix bug in both email status jobs to scan all bouncing emails not just first 250` forcing to retrieve older than a default limit?

Reply:
Yes - that's what the API docs say, and neither (existing) implementation does that.

This is just a generator to lazily retrieve all the email notifications that match the filter provided by the caller